elastic/kibana
# [EDR Workflows] The host isolation exception tab is hidden on the basic license if no artifacts

This PR updates how the Host Isolation Exceptions tab is displayed based on the user’s permissions and license. The tab is always visible to platinum+ users. For lower-tier licenses, a check is performed: if a user has previously defined host isolation exceptions, they will see the tab and be able to view or remove existing exceptions. If they haven’t, the tab will be hidden, and the functionality will be inaccessible.

Previously, even if a user didn’t have access to host isolation exceptions, they could still see and enter the Host Isolation Exceptions tab.

To test locally:
ESS:
1. Start ES + Kibana the regular way, with the default `trial` license. 
2. Add HIE
3. Downgrade license (https://github.com/elastic/pzl-es-tools) 
4. Verify that the license had been downgraded

Serverless:
1. Start Serverless ES `yarn es serverless --clean --teardown --kill -E xpack.security.authc.api_key.enabled=true -E http.host=0.0.0.0 --projectType security` 
2. Start Serverless Kibana `yarn serverless-security` 
3. Add HIE
4. Modify `config/serverless.security.yml` to security and endpoint essential
5. Wait for Kibana to reload

ESS:

https://github.com/user-attachments/assets/75527af7-9d06-4da7-9e86-6ce6b22ac147

Serverless:

https://github.com/user-attachments/assets/e89bd642-9e99-4a22-8b42-5997f7333ea6
